contract/src/main.rs

```diff
@@ -16,193 +19,139 @@ use types::{
-        CLType::Result {
-            ok: Box::new(CLType::String),
-            err: Box::new(CLType::String),
-        },
+    entry_points.add_entry_point(EntryPoint::new(
+        String::from("create"),
+        vec![
+            Parameter::new("uref", CLType::String),
+            Parameter::new("dict_key", CLType::String),
+            Parameter::new("dict_value", CLType::Option(Box::new(CLType::String))),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
     ));
-    entry_points.add_entry_point(endpoint("store_byte_array", CLType::ByteArray(3)));
-    entry_points.add_entry_point(endpoint(
-        "store_map",
-        CLType::Map {
-            key: Box::new(CLType::String),
-            value: Box::new(CLType::Option(Box::new(CLType::String))),
-        },
+    entry_points.add_entry_point(EntryPoint::new(
+        String::from("read"),
+        vec![
+            Parameter::new("uref", CLType::String),
+            Parameter::new("dict_key", CLType::String),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
     ));
-    entry_points.add_entry_point(endpoint(
-        "store_tuple3",
-        CLType::Tuple3([
-            Box::new(CLType::PublicKey),
-            Box::new(CLType::Option(Box::new(CLType::String))),
-            Box::new(CLType::U512),
-        ]),
+    entry_points.add_entry_point(EntryPoint::new(
+        String::from("update"),
+        vec![
+            Parameter::new("uref", CLType::String),
+            Parameter::new("dict_key", CLType::String),
+            Parameter::new("dict_value", CLType::Option(Box::new(CLType::String))),
+        ],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
     ));
-    entry_points.add_entry_point(endpoint(
-        "store_tuple2",
-        CLType::Tuple2([Box::new(CLType::String), Box::new(CLType::U512)]),
+    entry_points.add_entry_point(EntryPoint::new(
+        String::from("delete"),
+        vec![Parameter::new("uref", CLType::String)],
+        CLType::Unit,
+        EntryPointAccess::Public,
+        EntryPointType::Contract,
     ));
-
     let (contract_hash, _) =
         storage::add_contract_version(contract_package_hash, entry_points, Default::default());
-    runtime::put_key("kvstorage_contract", contract_hash.into());
+    runtime::put_key("kvstorage_contract_hash", contract_hash.into());
     let contract_hash_pack = storage::new_uref(contract_hash);
-    runtime::put_key("kvstorage_contract_hash", contract_hash_pack.into())
+    runtime::put_key("kvstorage_contract_hash_wrapped", contract_hash_pack.into())
 }
 
-fn read_and_store<T: CLTyped + FromBytes + ToBytes>() {
-    let name: String = runtime::get_named_arg("name");
-    let value: T = runtime::get_named_arg("value");
-    set_key(name.as_str(), value);
+fn create_key<T: ToBytes + CLTyped>(name: &str, dict_name: &str, dict_value: T) {
```

Comment:
You only allow String in the entry point so there is no need to make any of the function arguments generic